packages/babel-plugin/src/utils/evaluate-expression.tsx

```diff
@@ -205,6 +206,90 @@ const traverseFunction = (expression: t.Function, meta: Metadata) => {
   return createResultPair(value as t.Expression, updatedMeta);
 };
 
+/**
+ * Will find the function node for the call expression and wrap an IIFE around it (to avoid name collision)
+ * and move all the parameters mapped to passed arguments in the IIFE's scope (own scope in this case).
+ * It will also set own scope path so that when we recursively evaluate any node,
+ * we will look for its binding in own scope first, then parent scope.
+ *
+ * @param expression Expression we want to interrogate.
+ * @param meta Meta data used during the transformation.
+ */
+const traverseCallExpression = (expression: t.CallExpression, meta: Metadata) => {
+  const callee = expression.callee;
+  let value: t.Node | undefined | null = undefined;
+  let updatedMeta: Metadata = meta;
+
+  if (t.isExpression(callee)) {
```

Comment:
can we give some examples throughout this function of what each if statements callee/node would look like (so we don't forget later) please